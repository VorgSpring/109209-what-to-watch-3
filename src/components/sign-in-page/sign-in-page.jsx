import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {Redirect} from 'react-router-dom';
import {Header} from '../header/header.jsx';
import {SingInFormContainer} from '../sign-in-form/sign-in-form.jsx';
import {Footer} from '../footer/footer.jsx';
import {isAuthorizationSelector} from '../../selectors/authorization/authorization';
import {RoutePaths} from '../../constants/route-paths';

export const SingInPage = ({isAuthorizationRequired}) => (
  isAuthorizationRequired ? (
    <div className='user-page'>
      <Header user />

      <div className='sign-in user-page__content'>
        <SingInFormContainer />
      </div>

      <Footer />
    </div>
  ) : (
    <Redirect to={RoutePaths.MAIN} />
  )
);

SingInPage.propTypes = {
  isAuthorizationRequired: PropTypes.bool.isRequired
};

const mapStateToProps = (state) => ({
  isAuthorizationRequired: isAuthorizationSelector(state)
});

export const SingInPageContainer = connect(mapStateToProps)(SingInPage);
