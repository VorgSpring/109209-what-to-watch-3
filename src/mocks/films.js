export const films = [
  {
    id: 1,
    bgImage: `/bg-image`,
    name: `Fantastic Beasts: The Crimes of Grindelwald`,
    poster: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    previewImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    genre: `Crime`,
    released: 2000,
    preview: [
      {
        src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
        type: `video/mp4`
      },
      {
        src: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
        type: `video/webm`
      }
    ],
    isFavorite: false,
    rating: 1,
    director: `blah`,
    scoresCount: 3,
    description: `blah`,
    starring: [
      `blah`, `blah-blah`
    ],
    runTime: 123
  },
  {
    id: 2,
    bgImage: `/bg-image`,
    name: `Bohemian Rhapsody`,
    poster: `img/bohemian-rhapsody.jpg`,
    previewImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    genre: `Romance`,
    released: 2000,
    preview: [
      {
        src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
        type: `video/mp4`
      },
      {
        src: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
        type: `video/webm`
      }
    ],
    isFavorite: false,
    rating: 1,
    director: `blah`,
    scoresCount: 3,
    description: `blah`,
    starring: [
      `blah`, `blah`
    ]
  },
  {
    id: 3,
    bgImage: `/bg-image`,
    name: `Macbeth`,
    poster: `img/macbeth.jpg`,
    previewImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    genre: `Horror`,
    released: 2000,
    preview: [
      {
        src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
        type: `video/mp4`
      },
      {
        src: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
        type: `video/webm`
      }
    ],
    isFavorite: false,
    rating: 1,
    director: `blah`,
    scoresCount: 3,
    description: `blah`,
    starring: [
      `blah`, `blah`
    ]
  },
  {
    id: 4,
    bgImage: `/bg-image`,
    name: `Aviator`,
    poster: `img/aviator.jpg`,
    previewImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    genre: `Dramas`,
    released: 2000,
    preview: [
      {
        src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
        type: `video/mp4`
      },
      {
        src: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
        type: `video/webm`
      }
    ],
    isFavorite: false,
    rating: 1,
    director: `blah`,
    scoresCount: 3,
    description: `blah`,
    starring: [
      `blah`, `blah`
    ]
  },
  {
    id: 5,
    bgImage: `/bg-image`,
    bgColor: `#fff`,
    name: `We need to talk about Kevin`,
    poster: `img/we-need-to-talk-about-kevin.jpg`,
    previewImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    genre: `Documentary`,
    released: 2000,
    preview: [
      {
        src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
        type: `video/mp4`
      },
      {
        src: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
        type: `video/webm`
      }
    ],
    isFavorite: false,
    rating: 1,
    director: `blah`,
    scoresCount: 3,
    description: `blah`,
    starring: [
      `blah`, `blah`
    ],
    runTime: 123
  },
];

/* eslint-disable camelcase */
export const filmByServer = {
  id: 1,
  name: `The Grand Budapest Hotel`,
  poster_image: `img/the-grand-budapest-hotel-poster.jpg`,
  preview_image: `img/the-grand-budapest-hotel.jpg`,
  background_image: `img/the-grand-budapest-hotel-bg.jpg`,
  background_color: `#ffffff`,
  video_link: `https://some-link`,
  preview_video_link: `https://some-link`,
  description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
  rating: 8.9,
  scores_count: 240,
  director: `Wes Andreson`,
  starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`],
  run_time: 99,
  genre: `Comedy`,
  released: 2014,
  is_favorite: false,
};
