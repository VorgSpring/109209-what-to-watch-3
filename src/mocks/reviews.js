import {review} from './review';

export const reviews = {
  odd: [
    review
  ],
  even: [
    review
  ]
};
