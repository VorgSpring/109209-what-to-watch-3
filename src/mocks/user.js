export const user = {
  id: 1,
  email: `Oliver.conner@gmail.com`,
  name: `Oliver.conner`,
  avatarUrl: `img/1.png`
};

/* eslint-disable camelcase */
export const userByServer = {
  id: 1,
  email: `Oliver.conner@gmail.com`,
  name: `Oliver.conner`,
  avatar_url: `img/1.png`
};
