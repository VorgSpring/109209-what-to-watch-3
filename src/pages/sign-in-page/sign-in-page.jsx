import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {Redirect} from 'react-router-dom';
import {Header} from '../../components/header/header.jsx';
import {SingInFormContainer} from '../../components/sign-in-form/sign-in-form.jsx';
import {Footer} from '../../components/footer/footer.jsx';
import {isAuthorizationSelector} from '../../selectors/authorization/authorization';
import {RoutePaths} from '../../constants/route-paths';

export const SingInPage = ({isAuthorizationRequired}) => (
  isAuthorizationRequired ? (
    <div className='user-page'>
      <Header isMovieCard={false} />

      <div className='sign-in user-page__content'>
        <SingInFormContainer />
      </div>

      <Footer />
    </div>
  ) : (
    <Redirect to={RoutePaths.MAIN} />
  )
);

SingInPage.propTypes = {
  isAuthorizationRequired: PropTypes.bool.isRequired
};

const mapStateToProps = (state) => ({
  isAuthorizationRequired: isAuthorizationSelector(state)
});

export const SingInPageContainer = connect(mapStateToProps)(SingInPage);
