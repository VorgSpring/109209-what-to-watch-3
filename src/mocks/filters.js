import {ALL_GENRES_TYPE} from '../constants/films';

export const filters = {
  genre: ALL_GENRES_TYPE.ALL
};
