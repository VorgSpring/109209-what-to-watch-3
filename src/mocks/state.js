import {films} from './films';
import {filters} from './filters';
import {user} from './user';
import {review} from './review';

export const state = {
  films: {
    isLoading: false,
    isError: false,
    films
  },
  filters,
  authorization: {
    isLoading: false,
    isAuthorizationRequired: true,
    errorMessage: null
  },
  user,
  favorites: {
    isLoading: false,
    isError: false
  },
  promo: {
    isLoading: false,
    isError: false,
    film: films[0]
  },
  reviews: {
    isLoading: false,
    isError: false,
    reviews: {
      1: review
    }
  }
};
