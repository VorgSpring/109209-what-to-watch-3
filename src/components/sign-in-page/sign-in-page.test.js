import React from 'react';
import {BrowserRouter} from 'react-router-dom';
import {SingInPage} from './sign-in-page.jsx';
import {Provider} from 'react-redux';
import renderer from 'react-test-renderer';
import {state} from '../../mocks/state';
import configureStore from 'redux-mock-store';

describe(`SingInPage`, () => {
  it(`should renders correctly`, () => {
    const mockStore = configureStore([]);
    const store = mockStore(state);

    const tree = renderer.create(
        <Provider store={store}>
          <BrowserRouter>
            <SingInPage isAuthorizationRequired={false} />
          </BrowserRouter>
        </Provider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
