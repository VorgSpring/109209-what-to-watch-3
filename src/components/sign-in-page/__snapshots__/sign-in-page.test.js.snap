// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingInPage should renders correctly 1`] = `null`;
